### ISSUE TITLE
[out_datadog] on windows sends wrong timestamps, causing datadog to discard the log entries

### ISSUE THREAD

**Issue by jakila** [fluent/fluent-bit#2527]
## Bug Report

**Describe the bug**
On windows, the plugin out_datadog seems to send wrong timestamps to datadog servers. This causes datadog to discard the incoming log entries.
The exact same configuration and version of fluent-bit on Ubuntu 20.04 works fine, i.e. the log entries are visible on datadog.

**To Reproduce**
i am using the following configuration:
```
[INPUT]
    Name   dummy
    Tag    dummy.log

[OUTPUT]
    Name   stdout
    Match  *
    Format  json

[OUTPUT]
    Name              datadog
    Match             *
    Host              35.241.40.151
    compress          false
    apikey            <api-key removed>
```
using this configuration file with fluent-bit on windows outputs the following on the console:

```
[2020/09/04 17:47:22] [debug] [task] created task=000002839C4E3B80 id=0 OK
[{"date":1599234438.680823,"message":"dummy"},{"date":1599234439.68683,"message":"dummy"},{"date":1599234440.702288,"message":"dummy"},{"date":1599234441.695729,"message":"dummy"}]
[2020/09/04 17:47:22] [ info] [output:datadog:datadog.1] http://35.241.40.151, port=80, HTTP status=200 payload={}
[2020/09/04 17:47:22] [debug] [upstream] KA connection #500 to 35.241.40.151:80 is now available
[2020/09/04 17:47:22] [debug] [task] destroy task=000002839C4E3B80 (task_id=0)
```
and the following json is send to datadog-server (recorded using wireshark)
``` 
[{"timestamp":1506604569,"ddsource":"fluent-bi","service":"test-host","message":"dummy"},{"timestamp":1506605575,"ddsource":"fluent-bi","service":"test-host","message":"dummy"},{"timestamp":1506606590,"ddsource":"fluent-bi","service":"test-host","message":"dummy"},{"timestamp":1506607584,"ddsource":"fluent-bi","service":"test-host","message":"dummy"}]
```
Please notice the huge difference in the field "timestamp" between console output and json send to datadog.

**Expected behavior**
Using the exact same configuration on Ubuntu 20.04 with fluent-bit v1.5.4 sends the following json to datadog
```
[{"timestamp":1599230645991,"ddsource":"fluent-bit","service":"test-host","message":"dummy"},{"timestamp":1599230646991,"ddsource":"fluent-bit","service":"test-host","message":"dummy"},{"timestamp":1599230647991,"ddsource":"fluent-bit","service":"test-host","message":"dummy"},{"timestamp":1599230648991,"ddsource":"fluent-bit","service":"test-host","message":"dummy"},{"timestamp":1599230649991,"ddsource":"fluent-bit","service":"test-host","message":"dummy"}]
```
please notice the timestamps here are much closer to the one shown on the console (they are not exactly the same because i recorded them some minutes earlier).

**Your Environment**
* Version used: v1.5.4 Windows x64

**Additional context**
* i had trouble using `http-intake.logs.datadoghq.eu` in the `Host` field of out_datadog. So i am using ip adress instead. This seems unrelated to the issue
* i found the following workaround: renaming the `timestamp` field to something that is not parsed by the datadog servers, so that the ingress time is used instead. This is done by the field `json_date_key` of out_datadog

**Comment by fujimotos** [comment]
This is apparently an overflow issue. My guess is that this issue
is occurring due to `tm.tv_sec` being 32 bit on Windows

I'll take a closer look tomorrow; WFM.

**Comment by edsiper** [comment]
fixed by #2548
